src/idl_parser.cpp

```diff
@@ -555,7 +556,27 @@ CheckedError Parser::ParseType(Type &type) {
         return Error(
               "vector of union types not supported (wrap in table first).");
       }
-      type = Type(BASE_TYPE_VECTOR, subtype.struct_def, subtype.enum_def);
+      if (token_ == ':') {
+        if (!IsScalar(subtype.base_type)) {
+          return Error(
+                "arrays of non-scalar types not supported.");
+        }
+        NEXT();
+        if (token_ != kTokenIntegerConstant) {
+          return Error(
+                "length of fixed-length array must be an integer value.");
+        }
+        int64_t fixed_length = StringToInt(attribute_.c_str());
+        if (fixed_length < 1 || fixed_length > 0x7fff) {
```

Comment:
in reality the size needs to be even less in most cases, as a struct is stored in-line in a table, which needs to be addressable over 16bit vtable offsets, so an array of doubles can only have 2048 elements minus whatever table elements come after it. But we can't necessarily compute this exactly, since there may be multiple structs in the same table, etc.